### PR TITLE
Add ADR template and proposal

### PR DESCRIPTION
**Pull request by anishasthana**
- Add license for ADR repo
- Add Template for ADRs

### CODE REVIEW

File: PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## How
+
+We will keep each ADR in a short text file in Markdown format.
+
+We will keep ADRs in this repository, <https://github.com/project-codeflare/adr>.
+
+ADRs will be numbered sequentially and monotonically. Numbers will not be reused.
+
+If a decision is reversed, we will keep the old one around, but mark it as superseded. (It's still relevant to know that it was the decision, but is no longer the decision.)
```
Comment (KPostOffice) [inline PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md]:
```suggestion
If a decision is amended, we will keep the old one around, but mark it as superseded. (It's still relevant to know that it was the decision, but is no longer the decision.)
```
I think using amended here might be better. A decision doesn't have to be reversed in order to change.

File: PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+ADRs will be numbered sequentially and monotonically. Numbers will not be reused.
+
+If a decision is amended, we will keep the old one around, but mark it as superseded. (It's still relevant to know that it was the decision, but is no longer the decision.)
+
+We will use a format with just a few parts, so each document is easy to digest.
+
+## Approval Process / Flow
+
+TBD
```
Comment (KPostOffice) [inline PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md]: I feel like we should do something like "At least 1 approver and 1 week has passed" that way everyone should have a chance to comment on the ADR even if it is initially approved.

**Comment by asm582** [comment]
@anishasthana all projects should publish ADRs to this repo? should we create subfolders for each project?

File: PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md (hunk 2 above)
Comment (anishasthana) [reply PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md]: That sounds reasonable, but I worry 1 approver isn't enough.

**Comment by anishasthana** [comment]
> @anishasthana all projects should publish ADRs to this repo? should we create subfolders for each project?

As of right now I'm thinking we should keep them all in the same directory. I don't think we need to fragment things just yet... if later we find that the ADRs are getting too overloaded, we can break them out into separate folders.

**Comment by asm582** [comment]
LGTM

**Review by asm582 — DISMISSED** [review]

File: PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md (hunk 2 above)
Comment (KPostOffice) [reply PCF-ADR-0001-use-architecture-decision-records-for-project-codeflare.md]: 2? We should definitely keep it relatively low, we don't want to block decisions for too long.

**Review by KPostOffice — DISMISSED** [review]

**Review by KPostOffice — APPROVED** [review]

**Review by Maxusmusti — APPROVED** [review]
LGTM

**Pull request merged**